README.md

```diff
@@ -16,7 +16,26 @@ TODO
 
 ## Usage
 
-TODO
+Generate a peers ID:
+
+```go
+import (
+        "crypto/rand"
+
+        crypto "github.com/libp2p/go-libp2p-crypto"
+        peer "github.com/libp2p/go-libp2p-peer"
+)
+
+// Generate an identity keypair using go's cryptographic randomness source
+priv, pub, err := crypto.GenerateEd25519Key(rand.Reader)
+if err != nil { panic(err) }
+
+// A peers ID is the hash of its public key
+pid, err := peer.IDFromPublicKey(pub)
```

Comment:
In this case, you can actually call `peer.IDFromEd25519PublicKey` and embed the public key in the ID itself (we specify the hash function as the identity hash function).

Reply:
Cool! 

Sorry I'm new to libp2p/ipfs:
"IDFromEd25519PublicKey returns the Peer ID corresponding to Id25519 pk"
is there a significance to Id25519 that I've missed in the documentation somewhere?

For the usage section: Would you want me to switch the example from IDFromPublicKey to  IDFromEd25519PublicKey. Or have an example of both functions, or remove the example altogether?

Reply:
> Id25519 pk

I'm pretty sure that's a typo. It should be Ed25519 PK.

> usage section.

Maybe just mention this function in a comment? It's probably not necessary but it would be nice.

Reply:
Added peer.IDFromEd25519PublicKey to the code excerpt